opty/direct_collocation.py

```diff
@@ -1680,6 +1958,9 @@ def constraints(free):
             eom_con_vals = func(free_states, all_specified, all_constants,
                                 time_interval)
 
+            if all_specified.shape[0] == 1:
+                all_specified.squeeze()
```

Comment:
Should be safe to call squeeze() without the if statement, as it will only affect arrays with an empty dimension.